### PR TITLE
Enumerables

### PR DESCRIPTION
**Pull request by Tripple-A**
This is a ruby file containing methods to act like the ruby enumerabe methods.

### CODE REVIEW

File: enumerable_methods.rb [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  def my_all?(classification = nil)
+    test = true
+    checked = classification.class
+    my_each { |item| return false unless yield[item] == false } if block_given?
+    my_each { |item| return false unless item.class == classification } if checked == Class
+    my_each { |item| return false unless item =~ classification } if checked == Regexp
+    my_each { |item| return false unless item == classification } if [Integer, String].include?(checked)
+    my_each { |item| return false unless item } if !classification && !block_given?
+    test
+  end
```
Comment (wathigo) [inline enumerable_methods.rb]: **[REQUIRED]**  Returns true if the block never returns false or nil
```
enumerables my_all? returns true if the block never returns false or nil
     Failure/Error: let(:true_block) { proc { |num| num <= HIGHEST_VALUE } }
     
     NoMethodError:
       undefined method `<=' for nil:NilClass
     # ./spec/enumerables_spec.rb:59:in `block (4 levels) in <top (required)>'
     # ./enumerable_methods.rb:42:in `block in my_all?'
     # ./enumerable_methods.rb:9:in `my_each'
     # ./enumerable_methods.rb:42:in `my_all?'
     # ./spec/enumerables_spec.rb:62:in `block (3 levels) in <top (required)>'
```

File: enumerable_methods.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+  def my_any?(classification = nil)
+    test = false
+    my_each { |item| return true unless yield(item) == false } if block_given?
+    my_each { |item| return true unless item.class == classification } if classification.class == Class
+    my_each { |item| return true unless item =~ classification } if classification.class == Regexp
+    my_each { |item| return true unless item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| return true if item } if !classification && !block_given?
+    test
+  end
```
Comment (wathigo) [inline enumerable_methods.rb]: **[REQUIRED]**
* When a class is passed as an argument returns true if at least one of the collection is a member of such class
```
Failure/Error: expect(array.my_any?(Integer)).to be array.any?(Integer)
     
       expected true
            got false
     # ./spec/enumerables_spec.rb:119:in `block (4 levels) in <top (required)>'
```
* When a Regex is passed as an argument returns false if none of the collection matches the Regex.
```
Failure/Error: expect(words.my_any?(/z/)).to be words.any?(/z/)
     
       expected false
            got true
     # ./spec/enumerables_spec.rb:126:in `block (4 levels) in <top (required)>'
```
* When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern.
```
Failure/Error: expect(words.my_any?('cat')).to be words.any?('cat')
     
       expected false
            got true
     # ./spec/enumerables_spec.rb:133:in `block (4 levels) in <top (required)>'
```

File: enumerable_methods.rb [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+  def my_count(classification = nil)
+    counter = 0
+    my_each { |item| counter += 1 if item == classification } if [Integer, String].include?(classification.class)
+    my_each { |item| counter += 1 if yield(item) == true } if [Integer, String].include?(classification.class)
+    my_each { |item| counter += 1 if item.class == classification } if classification.class == Class
+    return length if !classification && !block_given?
+
+    counter
+  end
```
Comment (wathigo) [inline enumerable_methods.rb]: **[REQUIRED]** 
* Counts the number of items in enum that are equal to item if an argument is given.
```
Failure/Error: expect(array.my_count(LOWEST_VALUE)).to eq array.count(LOWEST_VALUE)
     
     LocalJumpError:
       no block given (yield)
     # ./enumerable_methods.rb:73:in `block in my_count'
     # ./enumerable_methods.rb:9:in `my_each'
     # ./enumerable_methods.rb:73:in `my_count'
     # ./spec/enumerables_spec.rb:187:in `block (3 levels) in <top (required)>'
```
* Counts the number of elements yielding a true value if a block is given.
```
Failure/Error: expect(array.my_count(&block)).to eq array.count(&block)
     
       expected: 54
            got: 0
     
       (compared using ==)
     # ./spec/enumerables_spec.rb:191:in `block (3 levels) in <top (required)>'
```

**Review by wathigo — CHANGES_REQUESTED** [review]
### Hi @Tripple-A  :wave: 
Great work implementing the changes requested by the other TSE.
There however a few things we didn't fix.
See the comments below :point_down:  

**After implementing the changes, submit another request for review.**

_Happy Coding :computer:  
Reviewed by: **Simon Wathigo**_

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: . Your project is approved and you may merge to development :+1: :+1: 

Keep up the good work :muscle: 
**Ahmed Mahfoudh**

**Pull request merged**